### ISSUE TITLE
Cross validation freeze the FBProphet on specific dataset

### ISSUE THREAD

**Issue by romanveaceslav** [facebook/prophet#1302]
I try to run forecasts many timeseries in a loop with cross validation.
In majority of cases it produces interesting results, but at specific timeseries Jupyter freezes.
Lanuching it in console helps identify that the issue is in cross validation. The intended model is 
logistic forecast with additional regressor. After the successful model fit (no errors) and during attempt to produce crossvalidation it prints several handred times: "Error evaluating model log probability: Non-finite gradient." after which freezes (left it for the night, nothing changes in 7 hours), even thow python process looks to fully utilize one CPU core during this time. Stopped with CTRL-C.  I would be happy to find a sollution that will return an exceptionafter a reasonable time (let's say 5 minutes) and let's things go on, so that I can continue doing other series forecast after exception handling.

Environment: Windows 10, Python 3.7.6, FBProphet 0.5, pystan 2.19.0, virtual environment for FBProphet (only FPProphet and its dependencies installed) in anaconda 2019.10.

Attached a source SQLite database (failed_data.db) for the data, a test script (test_failed.py) and log of its run on my computer.

A few words about the database: it contains one table (DATA_TO_FIT) with 3 fields: 'ds', 'y' and regressor for fit and cross validation. The reason I do provide it as a database is that I use a database for all my series and, as a curiosity, it fails only with data read from the database. If I read the database, save pandas to csv, read it back in pandas and do fit and cross validation that it still print, but much less "Error evaluating model log probability: Non-finite gradient." and then, finally produces the result, mean, it doe not freese. However, comparing dataframes from DB and csv there is a small difference in figures at 14-15 digit after the comma in the value of 'y' which explain this different behavior.
    
[issue_cross_validation.zip](https://github.com/facebook/prophet/files/4124356/issue_cross_validation.zip)

**Comment by bletham** [comment]
OK I ran the code. I didn't run into a freeze, but I suspect what is happening here is that the Stan optimizer is freezing. I have in the past seen rare instances of this happening, usually in situations with very bad model fits, often with very short time series.

This isn't a short time series, but I did notice that during one of the CV fits this printed out: `[...]fbprophet/forecaster.py:1251: RuntimeWarning: overflow encountered in exp`. There is an exp involved in computing the logistic trend, which could cause poor numerical behavior if the data are bad.

When I plot the model fit, it is clear that all of the data are _well_ below the cap. The cap is 1000; max(df['y']) is 19.82. If I set `cap` to a more reasonable value (I tried 25), then model fitting is much faster and I no longer get the warning about overflowing the exp. Could you see if that resolves the freeze for you?

In summary, the freeze is almost certainly due to numerical issues during fitting, and I think an excessively large value for cap is the source of the numerical issues here, since it is leading to bad model fits that overflow the exp.

**Comment by romanveaceslav** [comment]
Thank you. 
You are right, look like the cap value is a key here. For this dataset looks like there is a critical value of cap: when cap is 737 and bellow I do get some reasonable trend, above this value the trend is almost zero and do not correlate with the data change. Now, of cause, I can lower the capacity, but the difficulty will be to explain why I lower it bellow the phisical one and why such forecast shall be  credibile. Hm...
